src/StreamJsonRpc.Shared/JsonRpc.cs

```diff
@@ -679,23 +680,31 @@ protected virtual Task<ReturnType> InvokeCoreAsync<ReturnType>(int? id, string t
                         var attribute = (JsonRpcMethodAttribute) method.GetCustomAttribute(typeof(JsonRpcMethodAttribute));
                         if (attribute != null)
                         {
-                            List<string> names;
-                            if (!methodToAttributeMap.TryGetValue(method.Name, out names))
+                            string value;
+                            if (methodToAttributeMap.TryGetValue(method.Name, out value))
                             {
-                                names = new List<string>();
-                                methodToAttributeMap.Add(method.Name, names);
+                                if (!string.Equals(attribute.Name, value, StringComparison.Ordinal))
+                                {
+                                    throw new RemoteTargetException(string.Format(CultureInfo.CurrentCulture, Resources.ConflictingMethodNameAttribute, method.Name, nameof(JsonRpcMethodAttribute)));
+                                }
                             }
-
-                            if (!names.Contains(attribute.Name, StringComparer.Ordinal))
+                            else
                             {
-                                names.Add(attribute.Name);
-                            }
+                                methodToAttributeMap.Add(method.Name, attribute.Name);
```

Comment:
If you end up storing `MethodInfo`, I think you already have a test to make sure the system invokes an override method instead of a virtual one, so that should be useful to verify we don't have issues there.